api/v1alpha3/azuremachine_validation.go

```diff
@@ -32,3 +32,14 @@ func ValidateSSHKey(sshKey string, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+// ValidateUserAssignedIdentity validates the user-assigned identities list
+func ValidateUserAssignedIdentity(identityType VMIdentity, userAssignedIdenteties []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if identityType == VMIdentityUserAssigned && len(userAssignedIdenteties) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath, "The user-assigned identity provider ids must not be null or empty for 'UserAssigned' identity type."))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Required(fldPath, "must be specified for the 'UserAssigned' identity type"))
```

This will yield an error like `spec.whatever: Required value: must be specified for the 'UserAssigned' identity type`